File: MobileAppProjects/MobileApp/app/src/main/java/com/example/example2022/MainActivity.java
package com.example.example2022;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void convertCurrency(View v)
    {
        EditText dollarText = findViewById(R.id.dollarText);
        TextView euroText = findViewById(R.id.euroTV);

        if(!dollarText.getText().toString().equals(""))
        {
            float dollarValue = Float.parseFloat(dollarText.getText().toString());
            float euroValue = dollarValue * 0.85F;

            euroText.setText(String.format(Locale.ENGLISH, "%.2f", euroValue));
        }
        else
        {
            euroText.setText(R.string.no_value_string);
        }
    }
}

File: MobileAppProjects/app/src/main/java/com/example/flagquiz/MainActivity.kt
package com.example.flagquiz

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val startBtn: Button = findViewById(R.id.startBtn)
        val nameET: EditText = findViewById(R.id.nameET)

        startBtn.setOnClickListener {
            if (nameET.text.isEmpty())
            {
                Toast.makeText(this, "Please enter your name", Toast.LENGTH_LONG).show()
            }
            else
            {
                //create an instance of the questions activity
                val intent: Intent = Intent(this, QuestionsActivity::class.java)
                //send data via intent
                intent.putExtra(Constants.USERNAME, nameET.text.toString())
                //switch to the new activity
                startActivity(intent)
                //end the current activity
                finish()
            }
        }
    }
}